olp-cpp-sdk-core/include/olp/core/cache/DefaultCache.h

```diff
@@ -178,6 +178,27 @@ class CORE_API DefaultCache : public KeyValueCache {
    */
   bool Contains(const std::string& key) const override;
 
+  /**
+   * @brief Protect a list of keys from eviction.
+   * Keys or prefixes to protect single keys or entire catalogs, layers,
+   * version.
+   *
+   * @param keys list of keys and prefixes.
+   *
+   * @return True if the keys added to protected list; false otherwise.
+   */
+  bool Protect(const DefaultCache::KeyListType& keys);
+
+  /**
+   * @brief Release a list of keys from protection. The provided keys can be
+   * full keys or prefixes only.
+   *
+   * @param keys list of keys and prefixes.
+   *
+   * @return True if the keys removed from protected list; false otherwise.
+   */
```

Comment:
```suggestion
  /**
   * @brief Checks whether keys are removed from the protected list.
   *
   * The provided keys can be full keys or prefixes only.
   *
   * @param keys The list of keys and prefixes.
   *
   * @return True if the keys are removed from the protected list; false otherwise.
   */
```